start functionalizing make screenshots

kweav opened the pull request:
This PR begins implementing #145 by functionalizing `make_screenshots.R`, adding it to `ottrpal`. 

The approach I took was creating params based on the optparse options info in the script. I also carried over the functionality of the script. I'll leave a few comments with questions. Later stacked PRs will set up tests and  update the base_ottr image and update `render_all.yml` as described in #145

kweav commented on R/screenshot.R:
```diff
@@ -0,0 +1,60 @@
+#' @param git_pat default is NULL; required argument; a Git secret
+#' @param repo default is NULL; required argument; GitHub repository name, e.g., jhudsl/OTTR_Template
+#' @param output_dir default is "resources/chapt_screen_images"; Output directory where the chapter's screen images should be stored
+#' @param base_url default is NULL; rendered bookdown URL where screenshots are taken from, if NULL, the function will use the repo_name and and git_pat to find the base_url
```
Had to edit this description to change it from the original in the script. How is it?

cansavvy replied:
Looks like all the essentials are here! Only thing you could add if you wanted to be extra nice is a link to the page in GitHub about PATs: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

kweav commented on R/screenshot.R:
```diff
@@ -0,0 +1,60 @@
+#' A function to make screenshots from an OTTR bookdown website
+#' @description This function creates screenshots of course chapters that are stored in a created output directory 
+#' @param git_pat default is NULL; required argument; a Git secret
+#' @param repo default is NULL; required argument; GitHub repository name, e.g., jhudsl/OTTR_Template
```
What's the best way to communicate that these are required arguments, but by default NULL?

cansavvy replied:
I would just not have them be NULL if they are required because then it will automatically throw an error if no argument is provided. And docs wise the way you say its "required" is fine :+1:

kweav commented on R/screenshot.R:
```diff
@@ -0,0 +1,60 @@
+  }
+  
+  if (is.null(base_url)){
+    base_url <- cow::get_pages_url(repo_name = repo, git_pat = git_pat) #what if these arguments are still NULL/not supplied?
```
Do we want to make this cow function part of ottrpal?

cansavvy replied:
Yes! That's next up! It's mentioned in #137 but I didn't realize I didn't make it its own issue so there is one now: https://github.com/jhudsl/cow/issues/27

kweav commented on R/screenshot.R:
```diff
@@ -0,0 +1,60 @@
+  
+  message(paste("Image Chapter key written to: ", file.path(output_folder, "chapter_urls.tsv")))
+  
+}
```
Do I need to return something from this function?

cansavvy replied:
I think returning the file path to `chapter_urls.tsv` makes sense.

kweav commented:
pkgdown is failing because of trying to import cow as expected (and slacked)

cansavvy commented on R/screenshot.R:
```diff
@@ -0,0 +1,60 @@
+#' @description This function creates screenshots of course chapters that are stored in a created output directory 
+#' @param git_pat default is NULL; required argument; a Git secret
+#' @param repo default is NULL; required argument; GitHub repository name, e.g., jhudsl/OTTR_Template
+#' @param output_dir default is "resources/chapt_screen_images"; Output directory where the chapter's screen images should be stored
```
```suggestion
#' @param output_dir default is "resources/chapt_screen_images"; Output directory where the chapter's screen images should be stored. For OTTR courses, don't change this unless you've changed the downstream functions accordingly. 
```

cansavvy reviewed (approved):
This is looking good! :tada: Some minor comments!

Is the unit test for this coming in a next PR?

kweav commented:
> This is looking good! 🎉 Some minor comments!

Thanks! And thanks for the comments and suggestions! They're very helpful

> Is the unit test for this coming in a next PR?

Yes, the unit test will be the next PR. Can you point me to any examples/guides you want me to follow for creating a good unit test?

kweav commented:
For this PR my next steps will include addressing the suggestions/feedback here
And I will add a stacked PR that incorporates a unit test
And then I can also try to move the cow function into ottrpal in another stacked PR if you'd like?

kweav commented:
Latest commit addressed 
- adding the git pat link to the docs
- removing NULL as a default from the two required arguments
- tried to add an example of running the function (is that a good way of showing getting the secret?) 
- also have it return the file path now

kweav commented:
@cansavvy after merging #154 and removing the cow dependency, I was hoping that pkgdown wouldn't fail. But now it's failing because there is no `‘webshot2’` package 

```
Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
  there is no package called ‘webshot2’
```

kweav commented:
> @cansavvy after merging #154 and removing the cow dependency, I was hoping that pkgdown wouldn't fail. But now it's failing because there is no `‘webshot2’` package
> 
> ```
> Error in loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]) : 
>   there is no package called ‘webshot2’
> ```

We updated the `base_ottr` image in https://github.com/jhudsl/ottr-docker/pull/27 to include webshot2 and its various dependencies

kweav commented:
@cansavvy Given the R-CMD-check warnings about `check_git_repo` `get_git_auth` and `get_repo_info` not being global functions, I need to bring over those cow functions to `ottrpal` sooner rather than later, right? I know `check_git_repo` was next on my list, but should probably add the other two as well, right?

cansavvy commented:
> @cansavvy Given the R-CMD-check warnings about `check_git_repo` `get_git_auth` and `get_repo_info` not being global functions, I need to bring over those cow functions to `ottrpal` sooner rather than later, right? I know `check_git_repo` was next on my list, but should probably add the other two as well, right?

Yes this needs to be done. Probably transferring these at the same time makes sense. But probably in a next PR. :+1:

cansavvy commented:
Fixed the description file so now this is ready to merge!

Next PR can include more unit tests for this!
